### DISCUSSION TITLE
Get attention weight of heterogenous graph using GATv2

### DISCUSSION THREAD

**Discussion by Jilim97** [pyg-team/pytorch_geometric#8417] [Q&A]
Hello, I am having problem to get attention weight of heterogenous graph using GATv2 

  File "/opt/homebrew/lib/python3.10/site-packages/torch/fx/graph_module.py", line 662, in call_wrapped
    return self._wrapped_call(self, *args, **kwargs)
  File "/opt/homebrew/lib/python3.10/site-packages/torch/fx/graph_module.py", line 279, in __call__
    raise e.with_traceback(None)
TypeError: add(): argument 'input' (position 1) must be Tensor, not tuple

This is an error, and I use 2 GATv2 layers with 'add_self_loops=False' is layer command and 'return_attention_weights=True' in forward line.

If i use other layers (sage or Graph conv) with this data, it always run well and if i do not specifying return_attention_weights (as a default).
How can I solve it?

**Comment by rusty1s** [comment]
That is a current limitation. `to_hetero` sadly cannot handle `return_attention_weights`, and there exists no trivial way to support it. The only workaround would be to manually replicate what `to_hetero` is doing under the hood, e.g., replicating `GATConv` layers and applying them for each edge type individually.

**Reply by Jilim97** [reply]
Thank you for your reply. I can save my time no longer to find the method:)
